src/widget/woverview.cpp

```diff
@@ -565,6 +533,65 @@ void WOverview::paintEvent(QPaintEvent * /*unused*/) {
                 }
             }
 
+            // Draw range (loop)
+            for (auto&& markRange: m_markRanges) {
+                if (!markRange.active() || !markRange.visible()) {
+                    continue;
+                }
+
+                // Active mark ranges by definition have starts/ends that are not
+                // disabled.
+                const double startValue = markRange.start();
+                const double endValue = markRange.end();
+
+                const float startPosition = offset + startValue * gain;
+                const float endPosition = offset + endValue * gain;
+
+                if (startPosition < 0.0 && endPosition < 0.0) {
+                    continue;
+                }
+
+                if (markRange.enabled()) {
+                    painter.setOpacity(0.4);
+                    painter.setPen(markRange.m_activeColor);
+                    painter.setBrush(markRange.m_activeColor);
+                } else {
+                    painter.setOpacity(0.2);
+                    painter.setPen(markRange.m_disabledColor);
+                    painter.setBrush(markRange.m_disabledColor);
+                }
+
+                // let top and bottom of the rect out of the widget
+                if (m_orientation == Qt::Horizontal) {
+                    painter.drawRect(QRectF(QPointF(startPosition, -2.0),
+                                            QPointF(endPosition, height() + 1.0)));
+                } else {
+                    painter.drawRect(QRectF(QPointF(-2.0, startPosition),
+                                            QPointF(width() + 1.0, endPosition)));
+                }
+
+                // draw duration of range
+                if (markRange.showDuration()) {
+                    // TODO: replace with rate_ratio in PR #1765
+                    double rateRatio = 1.0 + m_pRateDirControl->get() * m_pRateRangeControl->get() * m_pRateSliderControl->get();
+                    QString duration = mixxx::Duration::formatTime((endValue - startValue)
+                            / m_trackSampleRateControl->get() / 2 / rateRatio);
```

Comment:
It might be worth to define a constant for the number of channels to not forget about it once the engine is rewritten.